go/test/endtoend/cluster/vtctl_process.go

```diff
@@ -71,13 +79,16 @@ func (vtctl *VtctlProcess) ExecuteCommandWithOutput(args ...string) (result stri
 		"-topo_implementation", vtctl.TopoImplementation,
 		"-topo_global_server_address", vtctl.TopoGlobalAddress,
 		"-topo_global_root", vtctl.TopoGlobalRoot}, args...)
+	if *isCoverage {
+		args = append([]string{"-test.coverprofile=" + getCoveragePath("vtctl-o-"+args[0]+".out"), "-test.v"}, args...)
+	}
 	tmpProcess := exec.Command(
 		vtctl.Binary,
 		args...,
 	)
 	log.Info(fmt.Sprintf("Executing vtctlclient with arguments %v", strings.Join(tmpProcess.Args, " ")))
 	resultByte, err := tmpProcess.CombinedOutput()
-	return string(resultByte), err
+	return filterResultWhenRunsForCoverage(string(resultByte)), err
```

Comment:
Nit:

As we are doing check for `isCoverage` already. Maybe we do the same for return results as well and only call filterResult method when isCoverage is true

Reply:
I check for `isCoverage` inside that function(`filterResultWhenRunsForCoverage`), as that function is being used from other place as well. I did not add that condition here, as I don't want to duplicate that condition in multiple places.